pagopa/io-app
# chore: [IOPID-2093] Add new get session endpoint

## Short description
The `getSession` call was parameterised in order to make it lighter, this API returns all tokens if it is not parameterised. 
In this PR, parameterisation handling has been added.
In order to have the `zendeskToken` when needed, an ad hoc handling has been created in the Support Section. 

**BUG FIX:** When the zendesk chat is opened if the user is logged in, there must be polling to check whether the session has expired or not. In the onboarding and first onboarding flow screens, polling was not present, so this fix was done 

## List of changes proposed in this pull request
- Added the possibility of passing parameters to the getSession
- By default, all tokens except the `zendeskToken` are passed as parameters to the getSession using this function `formatRequestedTokenString`
- Added retrieval of the `zendeskToken` in the Support Section using `getZendeskTokenSaga`
- Fix Zendesk Polling on onboarding and first onboarding flows

## Demo

### Dev Server

<p>

| First Onboarding | Onboarding | Logged User |
| - | - | - | 
| <video src="https://github.com/user-attachments/assets/c1082fc2-7fbf-4e20-ab87-3278f999641e"/> | <video src="https://github.com/user-attachments/assets/0b858784-7268-4b94-9ea6-cc76efefff1e"/> | <video src="https://github.com/user-attachments/assets/0bd81178-06c0-4989-9187-d9776516cf1d"/> |

</p>


### Production

<p>

| First Onboarding | Onboarding | Logged User |
| - | - | - | 
| <video src="https://github.com/user-attachments/assets/4e444788-8072-4d99-9a1a-d96bb04f4a23"/> | <video src="https://github.com/user-attachments/assets/22675eb2-cef4-42ec-9d03-bd006c2e8b05"/> | <video src="https://github.com/user-attachments/assets/6a74d7de-93ef-4e2f-8ea2-be38597368a9"/> |

</p>

 ### Bug Fix Polling

<p>

| dev server OLD | dev server NEW | Production OLD | PRODUCTION NEW |
| - | - | - | - |
| <video src="https://github.com/user-attachments/assets/cdd02515-5a31-4cad-a2d4-2aac0d5f3cfe"/> | <video src="https://github.com/user-attachments/assets/365f6142-00ec-4ab3-aa05-f1c97ba20f27"/> | <video src="https://github.com/user-attachments/assets/68f608f0-9c81-427f-a9fb-7f5db54052c3"/> | <video src="https://github.com/user-attachments/assets/a75db775-7d8d-4c0a-8047-f0492ce58e19"/> |

</p>


## How to test
Run the application or using the local environment, specifically [this branch](https://github.com/pagopa/io-dev-api-server/pull/417), or you can use the production environment and you can follow the same step that you can see in the Demo section (first onboarding, onboarding and legged user)
